### PR TITLE
Pu/issue709

### LINKED ISSUE #709: TODAY and NOW formulas return wrong value
Hello,

### Description
I am using TODAY and NOW formulas to get a current date but getting a date with a wrong day instead

I've checked coding and found out that these formulas return 
{
    **day: now.getDay()**
}

instead of 
{
    **day: now.getDate()**
}

These leads to getting a week day number instead of a real date

### Steps to reproduce
1. Set **=today()** value to A1 cell
2. Set **=day(a1)** value A2 cell
3. Check A2 value which is incorrect

### Demo
https://codesandbox.io/s/affectionate-banzai-zn7m1

* HyperFormula version: 0.6.2
* Browser Name and version: Chrome
* Operating System: Windows

### PR DESCRIPTION
**Pull request by izulin**
### Context

Fixes #709 

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1.
2.
3.

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation,
- [x] I described the modification in the CHANGELOG.md file.

### CODE REVIEW

File: test/interpreter/function-now.spec.ts [hunk 1]
```diff
@@ -9,17 +9,62 @@ function sleepFor( sleepDuration: number ){
...
 describe('Interpreter - function NOW', () => {
+  let originalNow: () => number
+
+  beforeEach(() => {
+    originalNow = Date.now
+    let cnt = 20
+    Date.now = () => {
+      cnt += 1
+      return Date.parse(`1985-08-16T03:45:${cnt}`)
+    }
```
Comment (wojciechczerniak) [inline test/interpreter/function-now.spec.ts:20]: A more elegant way will be and it's clear its a mock:

```suggestion
    dateMock = jest.fn().mockReturnValue(new Date('1985-08-16T03:45:30Z').getTime())
    Date.now = dateMock
```

I will get to NOW in the next comment

File: test/interpreter/function-now.spec.ts [hunk 2]
```diff
@@ -9,17 +9,62 @@ function sleepFor( sleepDuration: number ){
...
     const t1 = engine.getCellValue(adr('A1')) as number
+    expect(t1).toBeCloseTo(31275.1565856481)
     expect(engine.getCellValueDetailedType(adr('A1'))).toBe(CellValueDetailedType.NUMBER_DATETIME)
-    sleepFor(1000)
     engine.setCellContents(adr('A2'), null)
-    const t2 = engine.getCellValue(adr('A1')) as number
-    const delta = (t2-t1)*(24*60*60)
-    expect(delta).toBeCloseTo(1)
+    const delta = engine.getCellValue(adr('A1')) as number - t1
+    expect(delta*24*60*60).toBeGreaterThanOrEqual(1) //internals of the engine are also using Date.now(), so the value should be actually 4 or even more
```
Comment (wojciechczerniak) [inline test/interpreter/function-now.spec.ts:32]: Wouldn't it be better to have this test predictable?

```ts
dateMock.mockReturnValueOnce(new Date('1985-08-16T03:45:30Z').getTime())

const t1 = engine.getCellValue(adr('A1'))

dateMock.mockReturnValueOnce(new Date('1985-08-16T03:45:30Z').getTime())

engine.setCellContents(adr('A2'), null)
const t2 = engine.getCellValue(adr('A1'))
```

we will know exactly what time each `NOW()` returns and can make precise `expect`ation

**Review by wojciechczerniak — CHANGES_REQUESTED** [review]

File: test/interpreter/function-now.spec.ts (hunk 1 above)
Comment (izulin) [reply test/interpreter/function-now.spec.ts:20]: will it work for the multitude of testers that we have? (`jest`, `jasmine`, `karma`)?

Comment (wojciechczerniak) [reply test/interpreter/function-now.spec.ts:20]: true, it will not, I forgot about jasmine

Comment (izulin) [reply test/interpreter/function-now.spec.ts:20]: I don't think we should fight for some hybrid jest/jasmine mock tests and just go with what we have. But of course it is up to you

**Review by wojciechczerniak — APPROVED** [review]
As @izulin noticed correctly we can't use Jest mocks, they will fail in Jasmine

**Pull request merged**